File: 프로그래머스/1/42748. K번째수/K번째수.py
import heapq


def solution(array, commands):
    answer = []

    for i, j, k in commands:
        array_hq = array[i-1:j]
        heapq.heapify(array_hq)

        for _ in range(k-1):
            heapq.heappop(array_hq)

        answer.append(heapq.heappop(array_hq))

    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int i = 0; i<commands.size(); i++){
        int x = commands[i][0]-1;
        int y = commands[i][1]-1;
        int z = commands[i][2]-1;
        
        vector<int> tmp_list;
        
        for(int l = x; l<=y; l++){
            tmp_list.push_back(array[l]);
        }
        
        sort(tmp_list.begin(), tmp_list.end());
        answer.push_back(tmp_list[z]);
    }
    
    return answer;
}